.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,16 @@
+name: pre-commit
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+    - uses: pre-commit/action@v3.0.0
```

Comment:
As you know, we use make to run tests before commit, and I think it would be good if we use one entry point to test and apply style:
`make style`
`make check-style`
In this case, we can implement the pre-commit action on our side like this `github action` -> `make check-style` -> `pre-commit run`

Reply:
Added `make pre-commit`, it's more general name, if we will add another check like pylint, mypy.
`make check-style` i think it's no need, as code should be formatted. And i did not found direct flags to do it in pre-commit.